src/main/java/com/hms_networks/americas/sc/extensions/system/application/SCAppManagement.java

```diff
@@ -56,4 +58,98 @@ public static void disableAppAutoRestart() {
     Logger.LOG_DEBUG("Disabling next JVM run command.");
     RuntimeControl.configureNextRunCommand(jvmCommand);
   }
+
+  /**
+   * Checks if a valid WAN IP address has been assigned to the device.
+   *
+   * @return true if an initialized WAN IP address is found.
+   */
+  public static boolean deviceHasWanIP() {
+    boolean hasWanIp = false;
+    final String uninitializedWan = "0.0.0.0";
+
+    if (NetManager.getWanIp().equalsIgnoreCase(uninitializedWan)) {
+      hasWanIp = true;
+    }
+    return hasWanIp;
+  }
+
+  /**
+   * Waits for a number of minutes for a WAN IP to be establlshed if one does not already exist.
+   * After the given time has passed, returns false indicating no WAN IP is present.
+   *
+   * @param timeOutMinutes returns true if an initialized WAN IP is detected
+   */
+  public static boolean waitForWanIp(int timeOutMinutes) {
+
+    // There are 20 seconds in a wait period. 3 periods exist in a minute
```

Comment:
This comment refers to the 20 seconds in a period, which isn't yet defined.  See comment at line 128